src/Uno.UI.Runtime.Skia.Gtk/UI/Core/CursorExtensions.cs

```diff
@@ -5,45 +5,101 @@
 using System.Threading.Tasks;
 using Gdk;
 using Windows.UI.Core;
+using Uno.UI.Runtime.Skia.GTK.UI.Core;
 
 namespace Uno.UI.Runtime.Skia.GTK.Extensions
 {
 	internal static class CursorExtensions
 	{
 		public static CoreCursor ToCoreCursor(this Cursor cursor)
 		{
-			CoreCursorType GetCoreCursorType(CursorType? type) =>
-				type switch
-				{
-					CursorType.Arrow => CoreCursorType.Arrow,
-					CursorType.Cross => CoreCursorType.Cross,
-					CursorType.Hand1 => CoreCursorType.Hand,
-					CursorType.Hand2 => CoreCursorType.Hand,
-					CursorType.QuestionArrow => CoreCursorType.Help,
-					CursorType.Sizing => CoreCursorType.SizeAll,
-					CursorType.Watch => CoreCursorType.Wait,
-					_ => CoreCursorType.Arrow,
-				};
-			
-			return new CoreCursor(GetCoreCursorType(cursor?.CursorType), 0);
-		}
-
-		public static Cursor ToCursor(this CoreCursor coreCursor)
-		{
-			CursorType GetCursorType(CoreCursorType? coreCursorType) =>
-				coreCursorType switch
-				{
-					CoreCursorType.Arrow => CursorType.Arrow,
-					CoreCursorType.Cross => CursorType.Cross,
-					CoreCursorType.Hand => CursorType.Hand1,
-					CoreCursorType.Help => CursorType.QuestionArrow,
-					CoreCursorType.SizeAll => CursorType.Sizing,
-					CoreCursorType.Wait => CursorType.Watch,
-					_ => CursorType.Arrow,
-				};
+			CoreCursorType cursorType;
+			if (cursor == Cursors.Wait)
```

Comment:
Is there a reason you could not use the switch expression syntax ?

Reply:
Oops, I thought that the `switch` statement only supports "primitive" (enums, int, strings,...) types, and not class types like C++. Will fix it soon.

Reply:
Even on .NET 5, using the switch syntax requires a constant expression (a enum, int, string,... known at **compile time**). The cursors here are initialized during runtime.

Reply:
The switch syntax cannot be used here, your code in [WPF](https://github.com/trungnt2910/uno/blob/0971d5fb7b9756caedda6f7a70114142e9305cf2/src/Uno.UI.Runtime.Skia.Wpf/Extensions/CursorExtensions.cs#L10) does the same thing.